compiler/il/symbol/OMRResolvedMethodSymbol.cpp

```diff
@@ -863,13 +863,19 @@ OMR::ResolvedMethodSymbol::genAndAttachOSRCodeBlocks(int32_t currentInlinedSiteI
    // This exception edge should remain until OSR guards have been inserted, at which point it is no
    // longer needed
    //
-   bool genForOuterEntry = self()->comp()->isOutermostMethod() && self()->comp()->getHCRMode() == TR::osr;
-   if (genForOuterEntry)
+   if (self()->comp()->isOutermostMethod() && self()->comp()->getHCRMode() == TR::osr)
       {
+      // Add an exception edge to the OSR catch block from the first block
       TR::Block *OSRCatchBlock = osrMethodData->findOrCreateOSRCatchBlock(self()->getFirstTreeTop()->getNode());
       TR::Block *firstBlock = self()->getFirstTreeTop()->getEnclosingBlock();
       if (!firstBlock->hasExceptionSuccessor(OSRCatchBlock))
          self()->comp()->getFlowGraph()->addEdge(TR::CFGEdge::createExceptionEdge(firstBlock, OSRCatchBlock, self()->comp()->trMemory()));
+
+      // Create an OSR point with bytecode index 0
+      TR_ByteCodeInfo firstBCI = self()->getFirstTreeTop()->getNode()->getByteCodeInfo();
+      firstBCI.setByteCodeIndex(0);
```

Comment:
As assumptions may be invalidated during the method prologue and these may affect the JITed code representing bytecode index 0, it is necessary to allow for a transition before this is executed. There is no guarantee that the first treetop has this bytecode index correctly set, so this is done to be certain.